src/engine/renderer/tr_shade.cpp

```diff
@@ -1140,7 +1140,7 @@ static void Render_lightMapping( shaderStage_t *pStage )
 	}
 
 	// bind u_NormalMap
-	if ( !!r_normalMapping->integer || pStage->isHeightMapInNormalMap )
+	if ( !!r_normalMapping->integer || pStage->hasHeightMapInNormalMap )
```

Comment:
Shouldn't this just be `if ( r_normalMapping->integer )`?

After all, if normal mapping is disable, why bother setting stuff if `hasHeightMapInNormalMap` is true?

Reply:
> After all, if normal mapping is disable, why bother setting stuff if `hasHeightMapInNormalMap` is true?

We can load the normalmap+heightmap combined file to process the parallax, even if normal mapping is disabled but height mapping is enabled.